ModelBundle/Repository/NodeRepository.php

```diff
@@ -99,53 +99,59 @@ public function getSubMenu($nodeId, $nbLevel, $language, $siteId)
      *
      * @return mixed
      */
-    public function findOneCurrentlyPublished($nodeId, $language, $siteId)
+    public function findOnePublished($nodeId, $language, $siteId)
     {
         $qa = $this->createAggregationQueryBuilderWithSiteIdAndLanguage($siteId, $language);
         $filter = array(
-            'currentlyPublished' => true,
+            'status.publishedState' => true,
             'deleted' => false,
             'nodeId' => $nodeId,
         );
         $qa->match($filter);
-        $qa->sort(array('version' => -1));
 
         return $this->singleHydrateAggregateQuery($qa);
     }
 
     /**
-     * @param StatusableInterface $element
+     * @param string   $nodeId
+     * @param string   $language
+     * @param string   $siteId
+     * @param int      $version
      *
-     * @return StatusableInterface
+     * @return mixed
      */
-    public function findOneCurrentlyPublishedByElement(StatusableInterface $element)
+    public function findVersion($nodeId, $language, $siteId, $version)
```

Comment:
@alavieille : findVersionNoteDeleted to have the possibility to use a findVersion